gi_wrapper_cpp/dicom_reader.cpp

```diff
@@ -68,35 +78,48 @@ double* DicomReader::acquireImage(DataSet* dataSet, int& dims,
         exit(-1);
     }
 
-
-    double rescaleSlope = 1.0;
-    double rescaleIntercept = 0.0;
-    if (dataSet->bufferExists(TagId(tagId_t::DoseGridScaling_3004_000E), 0))
-        rescaleSlope = dataSet->getDouble(TagId(tagId_t::DoseGridScaling_3004_000E), 0);
-    else
-    {
-        if (dataSet->bufferExists(TagId(tagId_t::RescaleSlope_0028_1053), 0))
-            rescaleSlope = dataSet->getDouble(TagId(tagId_t::RescaleSlope_0028_1053), 0);
-
-        if (dataSet->bufferExists(TagId(tagId_t::RescaleIntercept_0028_1052), 0))
-            rescaleIntercept = dataSet->getDouble(TagId(tagId_t::RescaleIntercept_0028_1052), 0);
-    }
+    double rescaleSlope = im.GetSlope();
+    double rescaleIntercept = im.GetIntercept();
 
     double* array = new double[xNumber * yNumber * zNumber];
+    char* buffer = new char[im.GetBufferLength()];
+    int pixelSize = (int) im.GetPixelFormat().GetPixelSize();
     for (int k = 0; k < zNumber; k++)
     {
-        unique_ptr <imebra::Image> image(dataSet->getImage(k));
-        unique_ptr <ReadingDataHandlerNumeric> dataHandler(image->getReadingDataHandler());
         for (int j = 0; j < yNumber; j++)
         {
             for (int i = 0; i < xNumber; i++)
             {
-                array[(k * yNumber + j) * xNumber + i] = rescaleSlope * dataHandler->getDouble(j * xNumber + i) + rescaleIntercept;
+                if (pixelSize == 1) {
+                    _int8 currentValue;
```

Comment:
@kdbogusz please be aware that _int8 is not a standard fixed width integer type (if I'm not wrong it is exclusive to MSVC), please use int8_t from <cstdint> as it is listed as standardized fixed width integer type (please refer to https://en.cppreference.com/w/cpp/types/integer)